src/model/wallet.ts

```diff
@@ -457,13 +487,23 @@ export const signTypedDataMessage = async (
         version = 'v4';
       }
 
-      return {
-        result: signTypedData({
-          data: parsedData as TypedMessage<TypedDataTypes>,
-          privateKey: pkeyBuffer,
-          version: version.toUpperCase() as SignTypedDataVersion,
-        }),
-      };
+      // Hardware wallets
+      if (!wallet?.privateKey) {
```

Comment:
Is this a safe check? Like, 100% of the time, absence of `privateKey` means ledger?